docs/articles/rag-application-communication-system.md

```diff
@@ -0,0 +1,164 @@
+
+Interpolated datasets ran into a range of structural issues that proceed from the core nature of LLMs (a model trained at one time on a finite set of data):
+* **Specialization**: many professional resources are unlikely to be featured in pre-training data or with a very light exposure that result in high hallucination rates.
+* **Updatability**: a database can be continuously maintained in real time while LLMs are limited by their cutoff date.
+* **Explainability**: separating the search/memory part does circumvent to some extent the “black box” nature of LLMs.
+
+Instead, reformulation of few-document turned out to be one the most practical use cases for LLMs. The key properties were already built by design in word embeddings nearly a decade ago and have developed ever since. Cross-lingual transfer is surprisingly efficient: many models primarily trained in English can be successfully trained on downstream tasks in other languages. This extends to other forms of language transfer: from formal to informal, from standards to dialects, from generalist to specialized terminologies.
+
+The default paradigm for RAG emerged almost immediately after the release of chatGPT. One of the first systematic descriptions I’ve found is Simon Willison’s “(semantic search answers pattern)[https://simonwillison.net/2023/Jan/13/semantic-search-answers/]” from early January 2023 – search for texts in a semantic database, get the most relevant results, send this input to GPT-3 as an inspiration.
+
+Today, production RAG is increasingly a lot of flows and arrows and practitioners are confronted with the hard reality: there is no generic approach to RAG. Good applications need to anticipate a wide array of edge cases. There are simply just too many breaking points such as bad OCR, poor indexation, suboptimal tokenization, inconsistent document segment, unpredictable queries…
+
+The last mile problem is actually a great opportunity for RAG apps. Despite the eagerness of major AI actors to become encompassing platforms, the last mile won’t be easily solved. The right fit for a use case, a corpus, a specific user target is the hardest part of a RAG system, probably the only actual moat there is in applied generative AI.
+
+In turn, this means that the biggest issue for RAG is evaluation. Preliminary evaluations and benchmarks to select and swap models is not the hardest part — you could even do with synthetic data. The real challenge is evaluation in production of the different components of your system, taking into account their level of specialization. While we strive right now to automate everything, evaluation is a final residual of qualitative human assessment.
+
+## 2. Retrieval: back to future.
+As RAG was primarily formulated in an LLM/embedding context, it obfuscated the actual roots of the communication system it came to develop: information retrieval.
+
```

Comment:
"obfuscated" for whom? Don't ppl working on RAG well know that RAG is intended to solve information retrieval problems?

Reply:
yes so instead just going straight to the point: "Beyond LLMs, RAG is a new solution to an old issue: information retrieval"